source/geod24/LocalRest.d

```diff
@@ -303,22 +303,52 @@ public void sleep (Duration timeout)
         dg = If non-null, this delegate will be called when the timer fires
         periodic = Speficies if the timer fires repeatedly or only once
 
+    Return:
+        Timer Reference with stop function
+
 *******************************************************************************/
 
-public void setTimer (Duration timeout, void delegate() dg,
+public Timer setTimer (Duration timeout, void delegate() dg,
     bool periodic = false)
 {
     assert(scheduler !is null, "Cannot call this delegate from the main thread");
     assert(dg !is null, "Cannot call this delegate if null");
-    scheduler.schedule(
-    ()
+
+    Timer timer = Timer(timeout, dg, periodic);
+    scheduler.schedule(() { timer.run(); });
+    return timer;
+}
+
+/// Simple timer
+public struct Timer
+{
+    private Duration timeout;
+    private void delegate () dg;
+    private bool *periodic;
+
+    public this (Duration timeout, void delegate() dg, bool periodic)
+    {
+        this.timeout = timeout;
+        this.dg = dg;
+        this.periodic = new bool;
+        *this.periodic = periodic;
+    }
+
+    // Run an delegate after timeout
+    private void run ()
     {
         do
         {
             sleep(timeout);
             dg();
-        } while (periodic);
-    });
+        } while (*this.periodic);
+    }
+
+    // Delegate call stop
```

Comment:
This needs to be a ddoc comment with `///`.

```suggestion
    /// Stop the timer. The next time this timer's fiber wakes up
    /// it will exit the run() function.
```